addons/mail/models/mail_thread.py

```diff
@@ -2617,6 +2612,15 @@ def _notify_classify_recipients(self, recipient_data, model_name, msg_vals=None)
             for group_name, group_func, group_data in groups:
                 if group_func(recipient):
                     group_data['recipients'].append(recipient['id'])
+                    group_button_access = group_data.setdefault('button_access', {})
+                    temp_lang = self.env.user.lang
+                    self.env.user.lang = recipient['lang']
```

Comment:
as pta says, we are doing a write on the user (which may run lots of code), there is other hacks that are less invasive, for example:

```suggestion
                    context = {'lang': recipient['lang']}  # Use partner lang to translate mail subject below
                    ... # call the code that needs to be translated in other language
                    del context
```
or with self:

```suggestion
                    self = self.with_context(lang=recipient['lang'])
```
the _ method will use in the order:
>         # try, in order: context.get('lang'), kwargs['context'].get('lang'),
>         # self.env.lang, self.localcontext.get('lang'), request.env.lang

and `self.env.lang` is a shortcut to get the lang from `self.context.get('lang')`